File: Smallest Positive Integer that can not be represented as Sum - GFG/smallest-positive-integer-that-can-not-be-represented-as-sum.js
//{ Driver Code Starts
//Initial Template for javascript

"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin;
});

process.stdin.on("end", (_) => {
  inputString = inputString
    .trim()
    .split("\n")
    .map((string) => {
      return string.trim();
    });

  main();
});

function readLine() {
  return inputString[currentLine++];
}

function main() {
  let t = parseInt(readLine());
  let i = 0;
  for (; i < t; i++) {
    let n = parseInt(readLine());
    let input = readLine().split(' ').map(x => parseInt(x));
    let arr = [];
    for(let i=0;i<n;i++){
        arr.push(input[i]);
    }
    let obj = new Solution();
    let res = obj.smallestpositive(arr, n);
    console.log(res);
  }
}


// } Driver Code Ends


//User function Template for javascript


/**
 * @param {number[]} arr
 * @param {number} n
 * @returns {number}
 */

class Solution {
    smallestpositive(arr, n){
        let ans = 1;
        arr.sort((a,b)=>{return a-b});
        for(let val of arr){
            if(val > ans) return ans;
            else ans += val;
        }
        return ans;
    }
}

File: Smallest Positive Integer that can not be represented as Sum - GFG/smallest-positive-integer-that-can-not-be-represented-as-sum.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h> 
using namespace std; 


// } Driver Code Ends
//User function Template for C++
// m[i] -> minimum sum (integer) that is not present in sum subset of array upto position i 
class Solution
{   
public:
    long long smallestpositive(vector<long long> array, int n)
    { 
        using ll = long long;
        ll ans = 1; // ans ~ m[i-1]
        sort(array.begin(), array.end());
        for(int i = 0; i<n; i++){
            if(array[i] > ans) return ans;
            else{
                // maximum sum subset arr[i] + m[i-1] - 1;
                ans += array[i];
            }
        }
        return ans;
    } 
};


//{ Driver Code Starts.

int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        vector<long long> array(n);
        for (int i = 0; i < n; ++i)
        {
            cin>>array[i];
        }
        Solution ob;
        cout<<ob.smallestpositive(array,n)<<"\n";
    }
    return 0; 
} 

// } Driver Code Ends